cdap-data-fabric/src/main/java/co/cask/cdap/data2/registry/UsageDatasets.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.registry;
+
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.table.ConflictDetection;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.datafabric.dataset.DatasetsUtil;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.dataset2.DatasetManagementException;
+import co.cask.cdap.proto.Id;
+
+import java.io.IOException;
+
+/**
+ * Utility to create {@link UsageDataset}.
+ */
+public class UsageDatasets {
+  private static final Id.DatasetInstance USAGE_INSTANCE_ID =
+    Id.DatasetInstance.from(Constants.SYSTEM_NAMESPACE_ID, "usage.registry");
+
+  public static UsageDataset get(DatasetFramework framework) throws IOException, DatasetManagementException {
```

Comment:
I got the following exception when trying to instantiate from a handler in `AppFabricServer`:

```
2015-04-20 14:22:09,092 - ERROR [executor-15:c.c.c.d.d.d.DatasetsUtil@71] - Could NOT add dataset instance DatasetInstance{namespace=system, instance=usage.registry} of type co.cask.cdap.data2.registry.UsageDataset with props DatasetProperties{properties=conflict.level=NONE}
co.cask.cdap.data2.dataset2.DatasetManagementException: Failed to add instance usage.registry, details: Response code: 404, message:'Not Found', body: 'Cannot create dataset system.usage.registry: unknown type co.cask.cdap.data2.registry.UsageDataset'
	at co.cask.cdap.data2.datafabric.dataset.DatasetServiceClient.addInstance(DatasetServiceClient.java:142) ~[classes/:na]
	at co.cask.cdap.data2.datafabric.dataset.RemoteDatasetFramework.addInstance(RemoteDatasetFramework.java:126) ~[classes/:na]
	at co.cask.cdap.data2.datafabric.dataset.DatasetsUtil.createIfNotExists(DatasetsUtil.java:67) [classes/:na]
	at co.cask.cdap.data2.datafabric.dataset.DatasetsUtil.getOrCreateDataset(DatasetsUtil.java:54) [classes/:na]
	at co.cask.cdap.data2.registry.UsageDatasetUtil.getUsageDataset(UsageDatasetUtil.java:50) [classes/:na]
	at co.cask.cdap.gateway.handlers.UsageHandler$1.get(UsageHandler.java:53) [classes/:na]
	at co.cask.cdap.gateway.handlers.UsageHandler$1.get(UsageHandler.java:49) [classes/:na]
	at com.google.common.base.Suppliers$MemoizingSupplier.get(Suppliers.java:116) [guava-13.0.1.jar:na]
	at co.cask.cdap.gateway.handlers.UsageHandler.getAdapterStreamUsage(UsageHandler.java:123) [classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_51]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_51]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_51]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_51]
	at co.cask.http.HttpMethodInfo.invoke(HttpMethodInfo.java:85) [netty-http-0.11.0.jar:na]
	at co.cask.http.HttpDispatcher.messageReceived(HttpDispatcher.java:41) [netty-http-0.11.0.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70) [netty-3.6.6.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564) [netty-3.6.6.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791) [netty-3.6.6.Final.jar:na]
	at org.jboss.netty.handler.execution.ChannelUpstreamEventRunnable.doRun(ChannelUpstreamEventRunnable.java:43) [netty-3.6.6.Final.jar:na]
	at org.jboss.netty.handler.execution.ChannelEventRunnable.run(ChannelEventRunnable.java:67) [netty-3.6.6.Final.jar:na]
	at org.jboss.netty.handler.execution.OrderedMemoryAwareThreadPoolExecutor$ChildExecutor.run(OrderedMemoryAwareThreadPoolExecutor.java:314) [netty-3.6.6.Final.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_51]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_51]
	at java.lang.Thread.run(Thread.java:744) [na:1.7.0_51]
```